File: app/lib/cycle_ig/cycle_ig_share.dart
import 'package:menstrudel/cycle_ig/cycle_ig_snapshot.dart';

class CycleIgShare {
  final String id;
  final String fileUrl;
  final String manageToken;
  final String viewerLink;
  final String bareShlink;
  final String keyBase64Url;
  final String ciphertext;
  final DateTime expiresAt;
  final int maxUses;
  final CycleIgSnapshot snapshot;

  const CycleIgShare({
    required this.id,
    required this.fileUrl,
    required this.manageToken,
    required this.viewerLink,
    required this.bareShlink,
    required this.keyBase64Url,
    required this.ciphertext,
    required this.expiresAt,
    required this.maxUses,
    required this.snapshot,
  });
}

File: app/src/main/java/com/kollectivemobile/euki/cycleig/CycleIgShare.java
package com.kollectivemobile.euki.cycleig;

public class CycleIgShare {
    public final String id;
    public final String fileUrl;
    public final String manageToken;
    public final String viewerLink;
    public final String key;
    public final long exp;
    public final int maxUses;
    public final String bundleJson;
    public final String jwe;

    public CycleIgShare(String id, String fileUrl, String manageToken, String viewerLink, String key, long exp, int maxUses, String bundleJson, String jwe) {
        this.id = id;
        this.fileUrl = fileUrl;
        this.manageToken = manageToken;
        this.viewerLink = viewerLink;
        this.key = key;
        this.exp = exp;
        this.maxUses = maxUses;
        this.bundleJson = bundleJson;
        this.jwe = jwe;
    }
}
